### PR TITLE
bump k8s version to 1.23.15

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

bump k8s version to 1.23.15.   #4483 

**Note**: the runtime default change to `containerd`.

### CODE REVIEW

**Comment by Shelley-BaoYue** [comment]
need #4427 merged to solve the ci error of image build, because k8s 1.23 version requires go1.17 or greater.

**Comment by Shelley-BaoYue** [comment]
ci error need merge #4533  first.

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -149,6 +150,12 @@ func newEdged(enable bool, nodeName, namespace string) (*edged, error) {
 		return nil, fmt.Errorf("failed to construct kubelet dependencies")
 	}
 	edgedconfig.ConvertConfigEdgedFlagToConfigKubeletFlag(&edgedconfig.Config.TailoredKubeletFlag, &kubeFlags)
+	// Set Kubelet RegisterNode Parameter in KubeletConfiguration.
+	// The parameter `registerNode` has been migrated to Kubelet Configuration.
+	// `registerNode` in KubeletFlag will be retained for next version(1.13), and removed in 1.14 and later.
+	if edgedconfig.Config.TailoredKubeletConfig.RegisterNode == utilpointer.BoolPtr(false) {
+		kubeConfig.RegisterNode = false
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/edged/edged.go]: > The `kubeConfig` variable name is confusing， suggest rename to `kubeletConfig`

fixed.

[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply edge/pkg/edged/edged.go]: > if #4485 got merged, can we put this convert to ConvertEdgedKubeletConfigurationToConfigKubeletConfiguration

Yes, but if we put in the convert function, we need change the convert funtion input parameters with flag. I think it's better to  put it here and remove it next version.

**Comment by Shelley-BaoYue** [comment]
for new k8s tag contains 'prober not working', edged need adapt to this pr.  need merge #4470 first.

**Comment by Shelley-BaoYue** [comment]
PTAL  @fisherxu @wackxu @vincentgoat @gy95

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -242,7 +242,7 @@ jobs:
         run: |
           command -v ginkgo || go install github.com/onsi/ginkgo/v2/ginkgo@${{ env.GINKGO_VERSION }}
           go install sigs.k8s.io/kind@v0.11.1
```
Comment (gy95) [inline .github/workflows/main.yaml:244]: kind version also need updating
ref: https://github.com/kubernetes-sigs/kind/releases
maybe we should specify the kindest/node image tag when we call `kind create cluster` to use v1.23.15 k8s

File: edge/pkg/edged/edged.go [hunk 3]
```diff
@@ -332,10 +336,10 @@ func (e *edged) handlePodListFromEdgeController(content []byte, podCfg *config.P
 	for _, pod := range podLists {
 		if filterPodByNodeName(&pod, e.nodeName) {
 			pods = append(pods, &pod)
-			adds := &kubelettypes.PodUpdate{Op: kubelettypes.ADD, Pods: pods, Source: kubelettypes.ApiserverSource}
```
Comment (gy95) [inline edge/pkg/edged/edged.go:335]: this is a bug?

File: edge/pkg/edged/edged.go [hunk 4]
```diff
@@ -259,7 +267,7 @@ func MakeKubeClientBridge(kubeletDeps *kubelet.Dependencies) {
 	kubeletDeps.HeartbeatClient = client
 }
 
-func (e *edged) handlePod(op string, content []byte, podCfg *config.PodConfig) (err error) {
+func (e *edged) handlePod(op string, content []byte, updatesChan chan<- interface{}) (err error) {
```
Comment (gy95) [inline edge/pkg/edged/edged.go:270]: why modify edged.go like this? this is not related to k8s vendor bumping?

Comment (Shelley-BaoYue) [reply edge/pkg/edged/edged.go:270]: > why modify edged.go like this? this is not related to k8s vendor bumping?

refer： #4470

File: .github/workflows/main.yaml (hunk 2 above)
Comment (gy95) [reply .github/workflows/main.yaml:244]: it seems kindest/node:v1.23.15 not exist 👀 
the closest is kindest/node:v1.23.13

Comment (fisherxu) [reply .github/workflows/main.yaml]: `kindest/node:v1.23.13` is also fine :)
